external/tools/modify_imgui_for_gearsvk.ps1

```diff
@@ -0,0 +1,331 @@
+	# Output success of performed actions
+	$msg = "Comments about user texture support"
+	if ($appliedChanges["AddedFeature"] -eq 1 -And $appliedChanges["RemovedFeature"] -eq 1) {
+		$msg = "   [X] " + $msg
+		Write-Host $msg -ForegroundColor Green
+	}
+	else {
+		$msg = "   [ ] " + $msg
+		Write-Host $msg -ForegroundColor Red
+	}
+}
+
+
+##
+# Modify imgui vulkan impl source file (imgui_impl_vulkan.cpp)
+#
+# This adds:
+# - User texture comment to the implemented features list at the top:
+#	//  [x] Renderer: User texture binding. ImTextureID is used to store a handle to a Descriptorset.
+#
+# - Add code for texture descriptor set binding after "vkCmdSetScissor\(command_buffer, 0, 1, &scissor\);" in ImGui_ImplVulkan_RenderDrawData:
+# 	// Bind texture descriptor set stored as ImTextureID'
+# 	VkDescriptorSet desc_set[1] = { (VkDescriptorSet) pcmd->GetTexID() };'
+# 	vkCmdBindDescriptorSets(command_buffer, VK_PIPELINE_BIND_POINT_GRAPHICS, g_PipelineLayout, 0, 1, desc_set, 0, NULL);'
+#
+# This changes:
+# - The comment "// Bind pipeline and descriptor sets:" in ImGui_ImplVulkan_SetupRenderState to:
+#	// Bind pipeline
+#
+# - The line "io.Fonts->SetTexID((ImTextureID)(intptr_t)g_FontImage);" in ImGui_ImplVulkan_CreateFontsTexture for setting the font texture id to:
+#	io.Fonts->SetTexID((ImTextureID)g_DescriptorSet);
+#
+# This removes:
+# - User texture binding as missing feature comment at the top:
+#	//  [ ] Renderer: User texture binding
+#
+# - The unecessary font texture descriptor binding:
+#	VkDescriptorSet desc_set[1] = { g_DescriptorSet };
+#	vkCmdBindDescriptorSets(command_buffer, VK_PIPELINE_BIND_POINT_GRAPHICS, g_PipelineLayout, 0, 1, desc_set, 0, NULL);
+##
+
+"-> Modifying " + $imgui_impl_vulkan_cpp + ":"
+
+# Read imgui vulkan impl source
+$fileContent = Get-Content $imgui_impl_vulkan_cpp
+
+enum FunctionName {
+	SetupRenderState;
+	RenderDrawData;
+	CreateFontsTexture;
+	Undefined
+}
+$currentFunction = [FunctionName]::Undefined
+$aboutToEnterFunction = [FunctionName]::Undefined
+
+$appliedChanges = @{
+	[FunctionName]::SetupRenderState = 0
+	[FunctionName]::RenderDrawData = 0
+	[FunctionName]::CreateFontsTexture = 0
+}
+
+# Create an empty array and use it as the modified file
+$fileContentModified = @()
+
+# Header in file to communicate modification and prevent
+# multiple executions of this script
+$gearsVkHeader = "// Modified version for Gears-Vk"
+$fileContentModified += $gearsVkHeader
+$fileContentModified += '// Additions:'
+$fileContentModified += '// - User texture binding via ImTextureID'
+$fileContentModified += ''
+
+$fileAlreadyModified = 0
+$parentheseCounter = 0
+Foreach ($line in $fileContent)
+{
+	# Skip file if it already contains our header
+	if ($line -match $gearsVkHeader) {
+		Write-Host "   -> Already modified" -ForegroundColor Green
+		$fileAlreadyModified = 1
+		break
+	}
+
+	# Add user texture binding to implemented features list
+	if ($line -match "//\s*Missing\s*features:") {
```

Comment:
Powershells `-match` is case insensitive by default but we could use `-imatch` to make the intention more explicit

Reply:
Oh, really? Didn't know that (and it appears a bit strange to me ^^) 
Yes, `-imatch` would be cool, because then there can be no misunderstandings and it isn't at risk of changing its behavior, which `-match` might be (if the user changed the default behavior?).

Reply:
Yes, I discovered and googled it during regex testing because strings matched, when they shouldn't ^^
According to [About Comparison Operator](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_comparison_operators?view=powershell-7.1) all comparison operators are case-insensitive by default and I'm not sure if you can change this behaviour but you are right and I will change everything to `-imatch` because this default behaviour is a bit confusing for people who are not working with Powershell every day